packages/server/postgres/migrations/1709312768000_moveScheduledJob.ts

```diff
@@ -0,0 +1,59 @@
+import {FirstParam} from 'parabol-client/types/generics'
+import {Client} from 'pg'
+import {r} from 'rethinkdb-ts'
+import getPgConfig from '../getPgConfig'
+import connectRethinkDB from '../../database/connectRethinkDB'
+import getPgp from '../getPgp'
+
+export async function up() {
+  await connectRethinkDB()
+  const {pgp, pg} = getPgp()
+  const batchSize = 1000
+
+  const columnSet = new pgp.helpers.ColumnSet(
+    ['runAt', 'type', {name: 'orgId', def: null}, {name: 'meetingId', def: null}],
+    {table: 'ScheduledJob'}
+  )
+
+  const getNextData = async (leftBoundCursor: Date | undefined) => {
+    const startAt = leftBoundCursor || r.minval
+    const nextBatch = await r
+      .table('ScheduledJob')
+      .between(startAt, r.maxval, {index: 'runAt', leftBound: 'open'})
+      .orderBy({index: 'runAt'})
+      .limit(batchSize)
+      .run()
+    if (nextBatch.length === 0) return null
+    if (nextBatch.length < batchSize) return nextBatch
+    const lastItem = nextBatch.pop()
+    const lastMatchingRunAt = nextBatch.findLastIndex((item) => item.runAt !== lastItem!.runAt)
+    if (lastMatchingRunAt === -1) {
+      throw new Error(
+        'batchSize is smaller than the number of items that share the same cursor. Increase batchSize'
+      )
+    }
+    return nextBatch.slice(0, lastMatchingRunAt)
```

Comment:
-1 we should include `lastMatchingRunAt`
```
    return nextBatch.slice(0, lastMatchingRunAt + 1)
```
otherwise we might lose data. Assume a sequence [..., 4, 5, 5, 6], we would return [..., 4, 5] and start with the open (5, ...) next time, omitting this one element.

This sparks the idea that the migration should read the count first and verify it matches in the count in pg

Reply:
in future migrations I think I'll probably create a compound index that includes the timestamp + ID so we're guaranteed a unique ID that is also serial so items can still get updated while the migration is running

Reply:
@Dschoordsch this is not correct, see: https://github.com/ParabolInc/parabol/pull/9502#discussion_r1510340193